app/server.go

```diff
@@ -781,6 +791,46 @@ func doSessionCleanup(s *Server) {
 	s.Store.Session().Cleanup(model.GetMillis(), SESSIONS_CLEANUP_BATCH_SIZE)
 }
 
+func doLicenseExpirationCheck(s *Server) {
+	s.FakeApp().LoadLicense()
+	license := s.License()
+
+	if license == nil {
+		mlog.Info("License cannot be found.")
+		return
+	}
+
+	if !license.IsPastGracePeriod() {
+		mlog.Info("License is not past the grace period.")
+		return
+	}
+
+	users, err := s.Store.User().GetSystemAdminProfiles()
+	if err != nil {
+		mlog.Error("Failed to get system admins for license expired message from Mattermost.")
+		return
+	}
+
+	//send email to admin(s)
+	for _, user := range users {
```

Comment:
We are going to send this message every day?

Reply:
Only once, after the license removal, the first check returns.

Reply:
:facepalm: yes, you are totally right.